source/fuzz/fuzzer.cpp

```diff
@@ -192,6 +194,24 @@ bool Fuzzer::ApplyPassAndCheckValidity(
         }
       }
     }
+
+    // Check that distinct instructions have different unique ids.
+    std::unordered_map<uint32_t, opt::Instruction*> unique_ids;
```

Comment:
I think `std::unordered_set` is sufficient here.

Reply:
You're right, but the map is useful for debugging.  I extended the comment to explain this:

```
    // Check that all instructions have distinct unique ids.  We map each unique
    // id to the first instruction it is observed to be associated with so that
    // if we encounter a duplicate we have access to the previous instruction -
    // this is a useful aid to debugging.
```